website/content/middleware/rewrite.md

```diff
@@ -20,33 +20,59 @@ e.Pre(middleware.Rewrite(map[string]string{
 ```
 
 The values captured in asterisk can be retrieved by index e.g. $1, $2 and so on.
+Each asterisk will be non-greedy (translated to a capture group `(.*?)`) and if using
+multiple asterisk a trailing `*` will match the "rest" of the path.
+
+> Rewrite middleware should be registered via `Echo#Pre()` to get triggered before the router.
 
 ## Custom Configuration
 
-*Usage*
+### Usage
 
 ```go
 e := echo.New()
 e.Pre(middleware.RewriteWithConfig(middleware.RewriteConfig{}))
 ```
 
-## Configuration
+### Configuration
 
 ```go
 // RewriteConfig defines the config for Rewrite middleware.
-RewriteConfig struct {
-  // Skipper defines a function to skip middleware.
-  Skipper Skipper
+  RewriteConfig struct {
+    // Skipper defines a function to skip middleware.
+    Skipper Skipper
+
+    // Rules defines the URL path rewrite rules. The values captured in asterisk can be
+    // retrieved by index e.g. $1, $2 and so on.
+    Rules map[string]string `yaml:"rules"`
 
-  // Rules defines the URL path rewrite rules.
-  Rules map[string]string `yaml:"rules"`
-}
+    // RegexRules defines the URL path rewrite rules using regexp.Rexexp with captures
+    // Every capture group in the values can be retrieved by index e.g. $1, $2 and so on.
+    RegexRules map[*regexp.Regexp]string `yaml:"regex_rules"`
```

Comment:
I have not tried but I doubt `map[*regexp.Regexp]string`  can be (un)marshaled to yaml as `regexp.Regexp` does not implement needed interface.

Reply:
Removed the yaml tag.